packages/sw-lib/test/browser-unit/sw-unit/router.js

```diff
@@ -0,0 +1,126 @@
+importScripts('/node_modules/mocha/mocha.js');
+importScripts('/node_modules/chai/chai.js');
+importScripts('/node_modules/sw-testing-helpers/browser/mocha-utils.js');
+
+importScripts('/packages/sw-lib/build/sw-lib.min.js');
+importScripts('/packages/sw-routing/build/routing.min.js');
+
+/* global goog */
+
+const expect = self.chai.expect;
+self.chai.should();
+mocha.setup({
+  ui: 'bdd',
+  reporter: null,
+});
+
+describe('Test swlib.router', function() {
+  it('should be accessible goog.swlib.router', function() {
+    expect(goog.swlib.router).to.exist;
+  });
+
+  const badInput = [
+    {
+      capture: null,
+      errorName: 'unsupported-route-type',
+    },
+    {
+      capture: 123,
+      errorName: 'unsupported-route-type',
+    },
+    {
+      capture: true,
+      errorName: 'unsupported-route-type',
+    },
+    {
+      capture: false,
+      errorName: 'unsupported-route-type',
+    },
+    {
+      capture: {},
+      errorName: 'unsupported-route-type',
+    },
+    {
+      capture: [],
+      errorName: 'unsupported-route-type',
+    },
+    {
+      capture: '',
+      errorName: 'empty-express-string',
+    },
+  ];
+  badInput.forEach((badInput) => {
+    it(`should throw on adding invalid route: '${badInput}'`, function() {
+      let thrownError = null;
+      try {
+        goog.swlib.router.registerRoute(badInput.capture, () => {});
+      } catch (err) {
+        thrownError = err;
+      }
+
+      expect(thrownError).to.exist;
+      thrownError.name.should.equal(badInput.errorName);
+    });
+  });
+
+  it('should be able to register a valid express route', function() {
+    const date = Date.now();
+    const expressRoute = `/${date}/:id/test/`;
+    const exampleRoute = `/${date}/1234567890/test/`;
+
+    return new Promise((resolve, reject) => {
+      goog.swlib.router.registerRoute(expressRoute, () => {
+        // TODO: Checking ':id' makes it through to here.
+
```

Comment:
Things to resolve in the future.

I have left it for now as it requires some exploration and changes in sw-routing.